src/entry-point.tsx

```diff
@@ -60,6 +64,52 @@ export class LibraryController {
         this.reactor.raiseEvent(name, params);
     }
 
+    /**
+     * This method registers the callback function when the value of a given 
+     * named variable is requested. See PackageController.request method for 
+     * more information on each available variable name.
+     * 
+     * Note that if called more than once, this method replaces the original 
+     * callback function with the new value.
+     * 
+     * @param {string} variableName The name of the variable.
+     * @param {Function} callback The callback function to be invoked when a 
+     * variable of the given name is queried.
+     */
+    registerRequestHandler(variableName: string, callback: Function) {
+
+        // Storing the callback function in the property map.
+        this.requestHandler[variableName] = callback;
+    }
+
+    /**
+     * External code calls this method to obtain the value of the given named 
+     * variable. The execution of this method may or may not be asynchronous 
+     * depending on the corresponding handler registered in onRequest. Once 
+     * the value of the variable is obtained, the callback function is invoked 
+     * with the value.
+     * 
+     * Possible variable and their corresponding data types are as followed:
+     * 
+     *      packageState: string
+     * 
+     * @param {string} variableName The name of the variable whose value is to 
+     * be retrieved.
+     * @param {Function} callback The callback function to be invoked when the 
+     * variable value is obtained.
+     * @param {any[]} argsArray Zero or more argument values to be passed to the 
+     * subscribed handler for this named variable.
+     */
+    request(variableName: string, callback: Function, ...argsArray: any[]) {
+
+        // Invokve handler if there's one registered.
+        let requestHandler = this.requestHandler[variableName];
+        if (requestHandler && callback) {
+            let value = requestHandler(argsArray);
+            callback(value);
+        }
```

Comment:
Ah now I see why `callback` cannot be `null`. I think it should really be just like this (allowing callers to supply `null` for `callback` if they do not care about return value):

```js
// If the a request handler was not specified, the callback 
// function will still be invoked with a 'null' as result.
let value = null; 
if (requestHandler) {
    value = requestHandler(argsArray);
}
if (callback) {
    callback(value);
}
```